src/librairies/middlewares/validateId.middleware.ts

```diff
@@ -0,0 +1,33 @@
+import { Request, Response, NextFunction } from "express";
+import { isValidObjectId } from "mongoose";
+import BadRequestError from "../../config/error/bad.request.config";
+
+/**
+ * Middleware pour valider un ID dans la route
+ * @param entityName - Nom de l'entité (pour les messages d'erreur)
+ * @param paramName - Nom du paramètre dans la route (par défaut : "id")
+ */
+export const validateIdMiddleware =
+  (entityName: string, paramName: string = "id") =>
+  (req: Request, res: Response, next: NextFunction): void => {
+    try {
+      const id = req.params[paramName];
+
+      if (!isValidObjectId(id)) {
+        throw new BadRequestError({
+          message: `Invalid ${entityName} ID format`,
+          code: 400,
+          context: {
+            id,
+            entity: entityName,
+            operation: "validateId",
+            timestamp: new Date().toISOString(),
+          },
```

Comment:
le timestamp n'est pas nécessaire. le logger utilisé par le `errorHandlerMiddleware` nous permet déjà d'avoir le timestamp. et je pense qu'on ne devrait pas donner des message d'erreur trop explicites comme `Invalid ${entityName} ID format`.  ça donnes des piste à un attaquant qui cherche à savoir quels format d'id est correct pour faire du brute force par exemple. `Bad request` ou à la limite `Product not found` serait plus secure. si tu veux qu'on ait plus de détails, peut-être utiliser le logger